src/Features/CSharp/Portable/UpgradeProject/CSharpUpgradeProjectCodeFixProvider.cs

```diff
@@ -0,0 +1,91 @@
+        private const string CS8024 = nameof(CS8024); // error CS8024: Feature is not available in C# 3.  Please use language version X or greater.
+        private const string CS8025 = nameof(CS8025); // error CS8025: Feature is not available in C# 4.  Please use language version X or greater.
+        private const string CS8026 = nameof(CS8026); // error CS8026: Feature is not available in C# 5.  Please use language version X or greater.
+        private const string CS8059 = nameof(CS8059); // error CS8059: Feature is not available in C# 6.  Please use language version X or greater.
+
+        private static readonly ImmutableArray<string> s_diagnostics =
+            ImmutableArray.Create(CS8022, CS8023, CS8024, CS8025, CS8026, CS8059);
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = s_diagnostics;
+
+        public override string UpgradeThisProjectResource => CSharpFeaturesResources.Upgrade_this_project_to_csharp_language_version_0;
+        public override string UpgradeAllProjectsResource => CSharpFeaturesResources.Upgrade_all_projects_to_csharp_language_version_0;
+
+        public override IEnumerable<string> SuggestedVersions(ImmutableArray<Diagnostic> diagnostics)
+        {
+            var required = RequiredVersion(diagnostics);
+
+            var builder = ArrayBuilder<string>.GetInstance(1);
+
+            var generic = required <= LanguageVersion.Default.MapSpecifiedToEffectiveVersion()
+               ? LanguageVersion.Default // for all versions prior to current Default
+               : LanguageVersion.Latest; // for more recent versions
+
+            builder.Add(generic.ToDisplayString());
+
+            // also suggest the specific required version
+            builder.Add(required.ToDisplayString());
+
+            return builder.ToImmutableAndFree();
+        }
+
+        private LanguageVersion RequiredVersion(ImmutableArray<Diagnostic> diagnostics)
+        {
+            LanguageVersion max = 0;
+            foreach (var diagnostic in diagnostics)
+            {
+                if (diagnostic.Properties.TryGetValue(DiagnosticPropertyConstants.RequiredLanguageVersion, out string requiredVersion) &&
+                    LanguageVersion.Default.TryParseDisplayString(requiredVersion, out var required))
+                {
+                    max = max > required ? max : required;
+                }
+            }
+
+            return max;
+        }
+
+        public override Solution UpgradeProject(Solution solution, ProjectId projectId, string version)
+        {
+            var project = solution.GetProject(projectId);
+            var parseOptions = (CSharpParseOptions)project.ParseOptions;
+            LanguageVersion.Default.TryParseDisplayString(version, out var newVersion);
+
+            return solution.WithProjectParseOptions(projectId, parseOptions.WithLanguageVersion(newVersion));
```

Comment:
I'm in a language-specific class (`CSharpUpgradeProjectCodeFixProvider`) so I can directly use the APIs I need. 
Seems simpler than (1) get a language service, (2) convert the input language version to a string (so that the language service can parse it) and (3) invoke the language service API.